packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -239,7 +240,29 @@ class TestCommand extends FastFlutterCommand {
     if (build != 0) {
       throwToolExit('Error: Failed to build asset bundle');
     }
-    await writeBundle(fs.directory(fs.path.join('build', 'unit_test_assets')), assetBundle.entries);
+    if (_needRebuild(assetBundle.entries)) {
+      await writeBundle(fs.directory(fs.path.join('build', 'unit_test_assets')),
+          assetBundle.entries);
+    }
+  }
+  bool _needRebuild(Map<String, DevFSContent> entries) {
+    final File manifest = fs.file(fs.path.join('build', 'unit_test_assets', 'AssetManifest.json'));
+    if (manifest.existsSync()) {
+      final DateTime lastModified = manifest.lastModifiedSync();
+      final File pub = fs.file('pubspec.yaml');
+      if (pub.lastModifiedSync().isAfter(lastModified)) {
+        return true;
+      }
+      for (DevFSContent entry in entries.values) {
+        if (entry is DevFSFileContent) {
+          if (entry.isModified && entry.isModifiedAfter(lastModified)) {
```

Comment:
not sure if this is intended, the isModifiedAfter will return true if we have not access access the file stat before. no matter what lastModified is.

Reply:
This intended for the other usage of devfs, but is unfortunate in this case. I would add a comment documenting this behavior here

Reply:
added